sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputRequestThrottler.java

```diff
@@ -26,15 +27,28 @@
 
     private final AtomicReference<Double> availableThroughput;
     private final AtomicReference<Double> scheduledThroughput;
+    private final ReentrantReadWriteLock.WriteLock throughputWriteLock;
+    private final ReentrantReadWriteLock.ReadLock throughputReadLock;
 
     public ThroughputRequestThrottler(double scheduledThroughput) {
         this.availableThroughput = new AtomicReference<>(scheduledThroughput);
         this.scheduledThroughput = new AtomicReference<>(scheduledThroughput);
+        ReentrantReadWriteLock throughputReadWriteLock = new ReentrantReadWriteLock();
+        this.throughputWriteLock = throughputReadWriteLock.writeLock();
+        this.throughputReadLock = throughputReadWriteLock.readLock();
     }
 
-    public void renewThroughputUsageCycle(double scheduledThroughput) {
-        this.scheduledThroughput.set(scheduledThroughput);
-        this.updateAvailableThroughput();
+    public double renewThroughputUsageCycle(double scheduledThroughput) {
+        try {
+            this.throughputWriteLock.lock();
+            double throughputUsagePercentage = (this.scheduledThroughput.get() - this.availableThroughput.get()) / this.scheduledThroughput.get();
```

Comment:
can this number become negative ever? do we need to guard against negative number?

Reply:
This number will not become negative by theory.
1. When client setup the target Throughput/targetThroughputThreshold, we validate the value > 0. So scheduledThroughput should never be negative.
2. availableThroughput can be negative, but availableThroughput <= scheduledThroughput. So this.scheduledThroughput.get() - this.availableThroughput.get() should never be negative.